test/metabase/search/postgres/index_test.clj

```diff
@@ -204,3 +205,98 @@
   (testing "single quotes"
     (is (= "'you''re':*"
            (search-expr "you're")))))
+
+(defn ingest!
+  [model where-clause]
+  (#'search.ingestion/batch-update!
+   (#'search.ingestion/spec-index-reducible model where-clause)))
+
+(defn ingest-then-fetch!
+  [model entity-name]
+  (ingest! model [:= :this.name entity-name])
+  (t2/query-one {:select [:*]
+                 :from   [search.index/*active-table*]
+                 :where  [:and
+                          [:= :name entity-name]
+                          [:= :model model]]}))
+
+(deftest card-complex-ingestion-test
+  (search.tu/with-temp-index-table
+    (doseq [[model-type card-type] [["card" "question"] ["dataset" "model"] ["metric" "metric"]]]
+      (testing (format "simple %s" model-type)
+        (let [card-name (mt/random-name)]
+          (mt/with-temp [:model/Card {card-id :id} {:name card-name
+                                                    :type card-type}]
+            (is (=? {:model                    model-type
+                     :model_id                 card-id
+                     :official_collection      nil
+                     :database_id              (mt/id)
+                     :pinned                   false
+                     :view_count               0
+                     :collection_id            nil
+                     :dashboardcard_count      0
+                     :last_edited_at           nil
+                     :last_editor_id           nil
+                     :verified                 nil}
+                    (ingest-then-fetch! model-type card-name))))))
+
+      (testing (format "everything %s" model-type)
+        (let [search-term  (mt/random-name)
+              yesterday    (t/- (t/offset-date-time) (t/days 1))
+              two-days-ago (t/- yesterday (t/days 1))]
+          (mt/with-temp
+            [:model/Collection    {coll-id :id}      {:name            "My collection"
+                                                      ;; :official_collection = true
+                                                      :authority_level "official"}
+             :model/Card          {card-id :id}      {:name                search-term
+                                                      :type                card-type
+                                                      :query_type          "query"
+                                                      ;; :database_id = (mt/id)
```

Comment:
This comment doesn't seem necessary

Reply:
I would just stick to the counter intuitive ones 🤷 

Even the created_at => model_created_at stuff is obvious if you just look at how the variables are used.

Reply:
I just want to keep the convention so we know where stuffs come from, this one is obvious but the pined, last_edited_at stuff is not.